cpp/src/arrow/compute/exec.h

```diff
@@ -276,7 +276,7 @@ struct ExecValue {
   ArraySpan array = {};
   const Scalar* scalar = NULLPTR;
 
-  ExecValue(Scalar* scalar)  // NOLINT implicit conversion
+  ExecValue(const Scalar* scalar)  // NOLINT implicit conversion
```

Comment:
Isn't this a breaking change? Perhaps apply it in a separate PR?

Reply:
No, because a `Scalar*` can be accepted as `const Scalar*`, so this actually makes this constructor more broadly applicable.